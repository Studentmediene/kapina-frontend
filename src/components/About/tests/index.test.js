import React from 'react';
import { shallow } from 'enzyme';

import { About } from '../';

describe('<About />', () => {
  it('renders correctly', () => {
    const props = {
      about: '<p>about</p>',
      loading: false,
      error: false,
      loadAbout: jest.fn(),
    };

    const tree = shallow(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly while loading', () => {
    const props = {
      about: '<p>about</p>',
      loading: true,
      error: false,
      loadAbout: jest.fn(),
    };

    const tree = shallow(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly on error', () => {
    const props = {
      about: '<p>about</p>',
      loading: false,
      error: true,
      loadAbout: jest.fn(),
    };

    const tree = shallow(<About {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
