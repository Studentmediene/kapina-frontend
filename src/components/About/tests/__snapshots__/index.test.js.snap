// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>about</p>",
    }
  }
/>
`;

exports[`<About /> renders correctly on error 1`] = `
<div>
  Kunne ikke laste inn siden.
</div>
`;

exports[`<About /> renders correctly while loading 1`] = `<Loader />`;
